src/promiseQueue.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/**
+ * Function to throttle a list of promises.
+ *
+ * @param sourceQueue - The list of items to process.
+ * @param producer - The function to produce a promise from an item.
+ * @param concurrency - The number of promises to run at a time.
+ * @param queueResults - Whether to queue the results of the promises.
+ */
+export async function promisesQueue<T>(
+  sourceQueue: T[],
+  producer: (T) => Promise<T | T[]>,
+  concurrency: number,
+  queueResults = false
```

Comment:
@mshanemc it is used in mv function. Yes it can and yes I have :). Not being able to queue results defeats the ability to accomplish the dir traversal without recursion.